File: solutions/java/anagram/1/src/main/java/Anagram.java
import java.util.Arrays;
import java.util.List;
import java.util.ArrayList;

class Anagram {
    private String word;
    private String letters;
    
    public Anagram(String word){
        this.word = word;
        var charArr = word.toLowerCase().toCharArray();
        Arrays.sort(charArr);
        letters = new String(charArr);
    }

    public List<String> match(List<String> candidates){
        List<String> matches = new ArrayList<String>();

        for(var candidate: candidates){
            var candidateLetters = candidate.toLowerCase().toCharArray();
            Arrays.sort(candidateLetters);
            if (!this.word.equalsIgnoreCase(candidate) && this.letters.equals(new String(candidateLetters))) {
                matches.add(candidate);
			}
        }

        return matches;
    }
}

File: solutions/javascript/anagram/1/anagram.js
export const findAnagrams = (word, candidates) => {
  let sortedLetters = word.toLowerCase().split("").sort().join("");

  return candidates
    .filter((c) => c.toLowerCase().split("").sort().join("") == sortedLetters)
    .filter((c) => c.toLowerCase() != word.toLowerCase());
};
